Tests/xADGroup.Tests.ps1

```diff
@@ -37,159 +31,589 @@ Describe "xADGroup" {
-                Mock Move-ADObject { }
+
                 Set-TargetResource @testPresentParams;
+
                 Assert-MockCalled Move-ADObject -Scope It;
             }
-            It "Calls Remove-ADGroup when Ensure is Absent and group exists" {
+            
+            It "Resets group membership when 'Ensure' is 'Present' and 'Members' is incorrect" {
+                Mock Get-ADGroup { return [PSCustomObject] $fakeADGroup; }
+                Mock Set-ADGroup { }
+                Mock Get-ADGroupMember { return @($fakeADUser1, $fakeADUser2); }
+                Mock Add-ADGroupMember { }
+                Mock Remove-ADGroupMember { }
+
+                Set-TargetResource @testPresentParams -Members $fakeADuser1.SamAccountName;
+
+                Assert-MockCalled Remove-ADGroupMember -Scope It -Exactly 1;
+                Assert-MockCalled Add-ADGroupMember -Scope It -Exactly 1;
+            }
+            
+            It "Does not reset group membership when 'Ensure' is 'Present' and existing group is empty" {
+                Mock Get-ADGroup { return [PSCustomObject] $fakeADGroup; }
+                Mock Set-ADGroup { }
+                Mock Get-ADGroupMember { }
+                Mock Remove-ADGroupMember { }
+
+                Set-TargetResource @testPresentParams -MembersToExclude $fakeADuser1.SamAccountName;
+
+                Assert-MockCalled Remove-ADGroupMember -Scope It -Exactly 0;
+            }
+
+            It "Removes members when 'Ensure' is 'Present' and 'MembersToExclude' is incorrect" {
+                Mock Get-ADGroup { return [PSCustomObject] $fakeADGroup; }
+                Mock Set-ADGroup { }
+                Mock Get-ADGroupMember { return @($fakeADUser1, $fakeADUser2); }
+                Mock Remove-ADGroupMember { }
+
+                Set-TargetResource @testPresentParams -MembersToExclude $fakeADuser1.SamAccountName;
+
+                Assert-MockCalled Remove-ADGroupMember -Scope It -Exactly 1;
+            }
+            
+            It "Adds members when 'Ensure' is 'Present' and 'MembersToInclude' is incorrect" {
+                Mock Get-ADGroup { return [PSCustomObject] $fakeADGroup; }
+                Mock Set-ADGroup { }
+                Mock Get-ADGroupMember { return @($fakeADUser1, $fakeADUser2); }
+                Mock Add-ADGroupMember { }
+
+                Set-TargetResource @testPresentParams -MembersToInclude $fakeADuser3.SamAccountName;
+
+                Assert-MockCalled Add-ADGroupMember -Scope It -Exactly 1;
+            }
+            
+            It "Removes group when 'Ensure' is 'Absent' and group exists" {
                 Mock Get-ADGroup { return $fakeADGroup; }
                 Mock Remove-ADGroup { }
+            
                 Set-TargetResource @testAbsentParams;
+            
                 Assert-MockCalled Remove-ADGroup -Scope It;
             }
         } #end context Validate Set-TargetResource method
     
     } #end InModuleScope
+
```

Comment:
Style Nit pick: shouldn't have a newline between two closing curly braces.  This is not blocking.